cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/NamespacedDatasetFramework.java

```diff
@@ -97,8 +97,13 @@ public boolean hasInstance(Id.DatasetInstance datasetInstanceId) throws DatasetM
   }
 
   @Override
-  public boolean hasType(String typeName) throws DatasetManagementException {
-    return delegate.hasType(typeName);
+  public boolean hasSystemType(String typeName) throws DatasetManagementException {
```

Comment:
This class won't be removed. Instead, it will use actual namespaces to 'namespace' the dataset instance.